apache-kvrocks.yaml

```diff
@@ -0,0 +1,91 @@
+package:
+  name: apache-kvrocks
+  version: "2.11.1"
+  epoch: 0
+  description: Apache Kvrocks is a distributed key value NoSQL database that uses RocksDB as storage engine and is compatible with Redis protocol.
+  copyright:
+    - license: Apache-2.0
+
+environment:
+  contents:
+    packages:
+      - autoconf
+      - automake
+      - bash
+      - build-base
+      - busybox
+      - cmake
+      - coreutils
+      - libtool
+      - lld
+      - openssl-dev
+
+pipeline:
+  - uses: git-checkout
+    with:
+      repository: https://github.com/apache/kvrocks
+      tag: v${{package.version}}
+      expected-commit: 8d56734134918120520abdcec70dcf0f73db5492
+
+  - uses: cmake/configure
+    with:
+      opts: |
+        -DENABLE_OPENSSL=ON \
+        -DPORTABLE=1 \
+        -DCMAKE_BUILD_TYPE=Release \
+
+  - uses: cmake/build
+
+  - uses: cmake/install
+
+  # Install does not copy binary over, installing it manually
+  - runs: |
+      mkdir -p "${{targets.contextdir}}/usr/bin"
+      install -m755 "./output/kvrocks" "${{targets.contextdir}}/usr/bin/kvrocks"
+
+  - uses: strip
+
+subpackages:
+  - name: ${{package.name}}-compat
+    description: "Compatibility package for apache-kvrocks image"
+    pipeline:
+      - runs: |
+          mkdir -p "${{targets.contextdir}}/var/lib/kvrocks"
+          mkdir -p "${{targets.contextdir}}/kvrocks"
+          install -m644 ./LICENSE "${{targets.contextdir}}/kvrocks/"
+          install -m644 ./NOTICE "${{targets.contextdir}}/kvrocks/"
+          install -m644 ./licenses/LICENSE-* "${{targets.contextdir}}/kvrocks/"
+          install -m644 ./kvrocks.conf "${{targets.contextdir}}/var/lib/kvrocks/"
+    test:
+      pipeline:
+        - runs: |
+            # verify config file is provided
+            test -f /var/lib/kvrocks/kvrocks.conf
+            # verify LICENSE and NOTICE files are provided
+            test -f /kvrocks/NOTICE
+            test -f /kvrocks/LICENSE
+            # verify at least one third party license file provided
+            test -f /kvrocks/LICENSE-zlib.txt
+
+update:
+  enabled: true
+  github:
+    identifier: apache/kvrocks
+    strip-prefix: v
+
+test:
+  environment:
+    contents:
+      packages:
+        - valkey-cli
+        - wait-for-port
+  pipeline:
+    - runs: |
+        kvrocks &
```

Comment:
looks like here we can use test/daemon-check-output pipeline, see some logs in action and then do the valkey-cli invocations in post: block.